src/libsync/account.h

```diff
@@ -16,6 +16,7 @@
 #ifndef SERVERCONNECTION_H
 #define SERVERCONNECTION_H
 
+#include "userstatusjob.h"
```

Comment:
Yes, we could do that. I hate forward declarations (because the sometimes lead to interesting link errors), but I understand the benefit. My editor inserts the includes automatically.

Reply:
I like the `std::shared_ptr` more because I think more people will easily and faster understand the code because `std::shared_ptr` is part of the standard. For `QSharedPointer` you have to look up the documentation. What does @mgallien think?

Reply:
usage of forward declaration leads to two advantages

- faster compilation because headers have less includes in them
- less coupling because the interface of classes (as described in the headers) should use as less include as necessary top hide the internal implementation choices

I tend to prefer std smart pointers and would not be surprised that Qt would want remove their own ones to make it easier to mix code using Qt frameworks and code using only the STL library.

Reply:
Fixed